File: iphone/Maps/UI/PlacePage/PlacePageLayout/CopyLabel.m
#import "CopyLabel.h"

@implementation CopyLabel

- (instancetype)initWithFrame:(CGRect)frame {
  self = [super initWithFrame:frame];
  if (self) {
    self.userInteractionEnabled = YES;
  }
  return self;
}

- (BOOL)canBecomeFirstResponder {
  return YES;
}

- (BOOL)canPerformAction:(SEL)action withSender:(id)sender {
  return action == @selector(copy:);
}

- (void)copy:(id)sender {
  [UIPasteboard generalPasteboard].string = self.text;
}

@end

File: iphone/Maps/UI/PlacePage/PlacePageLayout/CopyLabel.swift
final class CopyLabel: UILabel {
  override var canBecomeFirstResponder: Bool {
    true
  }

  override func canPerformAction(_ action: Selector, withSender _: Any?) -> Bool {
    action == #selector(copy(_:))
  }

  override func copy(_: Any?) {
    UIPasteboard.general.string = text
  }
}
